File: leetcode/two_sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        number_dict = {}

        for i in range(len(nums)):
            comp = target - nums[i]
            if comp in number_dict:
                return [number_dict[comp], i]
            else:
                number_dict[nums[i]] = i

nums = [2, 7, 11, 15]
target = 9

r = Solution().twoSum(nums, target)
print(r)

File: leetcode/two_sum.cpp
#include <iostream>
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
  vector<int> twoSum(vector<int> &nums, int target) {
    unordered_map<int, int> hashmap;
    vector<int> result;
    for (int i = 0; i < nums.size(); i++) {
      int comp = target - nums[i];
      if (hashmap.find(comp) != hashmap.end()) {
        result.push_back(hashmap[comp]);
        result.push_back(i);

        return result;
      } else {
        hashmap[nums[i]] = i;
      }
    }
    return result;
  }
};

int main() {
  Solution s;
  vector<int> nums = {2, 7, 11, 15};
  int target = 9;

  vector<int> r = s.twoSum(nums, target);
  cout << "result: ";
  for (int i : r) {
    cout << i << " ";
  }

  cout << endl;

  return 0;
}
